pkg/apis/application/v1alpha2/app_catalog_types.go

```diff
@@ -0,0 +1,120 @@
+package v1alpha2
+
+import (
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/apiextensions/v3/pkg/crd"
+)
+
+const (
+	kindAppCatalog              = "AppCatalog"
+	appCatalogDocumentationLink = "https://docs.giantswarm.io/reference/cp-k8s-api/appcatalogs.application.giantswarm.io/"
+)
+
+func NewAppCatalogCRD() *v1.CustomResourceDefinition {
+	return crd.LoadV1(group, kindAppCatalog)
+}
+
+func NewAppCatalogTypeMeta() metav1.TypeMeta {
+	return metav1.TypeMeta{
+		APIVersion: SchemeGroupVersion.String(),
+		Kind:       kindAppCatalog,
+	}
+}
+
+// NewAppCatalogCR returns an AppCatalog Custom Resource.
+func NewAppCatalogCR() *AppCatalog {
+	return &AppCatalog{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				annotation.Docs: appCatalogDocumentationLink,
+			},
+		},
+		TypeMeta: NewAppCatalogTypeMeta(),
+	}
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories=common;giantswarm
+// +kubebuilder:storageversion
+// +k8s:openapi-gen=true
+// AppCatalog represents a catalog of managed apps. It stores general information for potential apps to install.
+// It is reconciled by app-operator.
+type AppCatalog struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              AppCatalogSpec `json:"spec"`
+}
+
+// +k8s:openapi-gen=true
+type AppCatalogSpec struct {
+	// Title is the name of the app catalog for this CR
+	// e.g. Catalog of Apps by Giant Swarm
+	Title       string `json:"title"`
+	Description string `json:"description"`
+	// +kubebuilder:validation:Optional
+	// +nullable
+	// Config is the config to be applied when apps belonging to this
+	// catalog are deployed.
+	Config AppCatalogSpecConfig `json:"config,omitempty"`
```

Comment:
```suggestion
	Config *AppCatalogSpecConfig `json:"config,omitempty"`
```

Config and the fields below are optional and we use omitempty. But we still get empty fields for YAML output which are ugly. This is because we don't use pointers for the structs.

Theo, added this for app CRs in https://github.com/giantswarm/apiextensions/pull/283 but in the end, we didn't implement it. As we'd need to check for nils in a lot of places.

Since this is a new CRD should we implement it? If we always use the key package from the app library we could add the nil handling there. WDYT?